src/Cache.ts

```diff
@@ -320,7 +319,6 @@ export class Cache {
 
         for (const section of sections) {
             if (
-                section.type === 'list' &&
                 section.position.start.line <= lineNumberTask &&
```

Comment:
Just dropping here, for info, the file I was using to explore the behaviour of this PR while reviewing the code.

[Callouts and Block Quotes.md](https://github.com/obsidian-tasks-group/obsidian-tasks/files/9269482/Callouts.and.Block.Quotes.md)

I will leave this comment unresolved for now, to remind me to commit this file after merging the PR.

- [ ] Commit [Callouts and Block Quotes.md](https://github.com/obsidian-tasks-group/obsidian-tasks/files/9269482/Callouts.and.Block.Quotes.md) - in the `Manual Testing` folder of the `Tasks-Demo` vault.

Reply:
A particular thing I did in this file was to include some things that looked like tasks but should not be, to make sure they were not now picked up as tasks.

Reply:
> Note: I get a lefthook error on the pre-commit hook when committing things to the "Manual Testing" folder due to the space in the folder name. Curious if you can repro that or if it is Windows-specific.
> Anyway, leaving this for you due to this issue.

Weird.

I suggest that if we are going to discuss this, we do it in a different issue, as this is quite big already 😄

Reply:
I will leave this unresolved, to remind me to add the file after this is merged.

Reply:
OK - resolving now so I can merge. Will add this file tomorrow.

Reply:
Hi @AnnaKornfeldSimpson FYI, I've added the sample markdown file `Callouts and Block Quotes.md` in #996.